tus/src/main/java/ch/cyberduck/core/tus/TusCapabilitiesResponseHandler.java

```diff
@@ -0,0 +1,66 @@
+package ch.cyberduck.core.tus;
+
+/*
+ * Copyright (c) 2002-2024 iterate GmbH. All rights reserved.
+ * https://cyberduck.io/
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ */
+
+import ch.cyberduck.core.io.HashAlgorithm;
+
+import org.apache.commons.lang3.StringUtils;
+import org.apache.http.HttpResponse;
+import org.apache.http.HttpStatus;
+import org.apache.http.client.ResponseHandler;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+public class TusCapabilitiesResponseHandler implements ResponseHandler<TusCapabilities> {
+    private static final Logger log = LogManager.getLogger(TusCapabilitiesResponseHandler.class);
+
+    @Override
+    public TusCapabilities handleResponse(final HttpResponse response) {
+        switch(response.getStatusLine().getStatusCode()) {
+            case HttpStatus.SC_NO_CONTENT:
+                if(log.isDebugEnabled()) {
+                    log.debug(String.format("Support for resumable uploads detected from %s", response));
+                }
+                final TusCapabilities tus = new TusCapabilities();
+                if(response.containsHeader(TusCapabilities.TUS_HEADER_RESUMABLE)) {
+                    tus.withVersion(response.getFirstHeader(TusCapabilities.TUS_HEADER_RESUMABLE).getValue());
```

Comment:
Would rather try to match a version in `Tus-Version` with the one we support.

Reply:
This is up to the caller [^1] 

[^1]: https://github.com/iterate-ch/cyberduck/pull/15788/files#diff-0ef80a78b96e7846613a681f29f6b14f6c2faf83a480a76e91a908bf8cfec80fR98

Reply:
Of course but not the point. The following `OPTIONS` response would not enable the TUS extension although the server supports version 1.0.0:
```
HTTP/1.1 204 No Content
Tus-Resumable: 1.1.0
Tus-Version: 1.1.0,1.0.0,0.2.2,0.2.1
Tus-Max-Size: 1073741824
Tus-Extension: creation,expiration
```